### ISSUE TITLE
Allow to add go2rtc as a network camera in Windows

### ISSUE THREAD

**Issue by felipecrs** [AlexxIT/go2rtc#1290]
Windows has this feature of network cameras:

![image](https://github.com/user-attachments/assets/9139587d-5d1e-4c08-9289-7de5b4b0cbdb)

It would be amazing if Windows could recognize go2rtc as a network camera.

But that would require go2rtc to implement the ONVIF WS-Discovery protocol:

https://learn.microsoft.com/en-us/windows-hardware/drivers/stream/network-cameras#discovery

Do you think that would be possible?

My use case is very niche:

When running Microsoft Teams under Azure Virtual Desktop, it will only recognize cameras that are "physically" connected to the client computer and not virtual cameras like OBS Studio or NVIDIA Broadcast.

With go2rtc running on Windows, I can input the camera with `ffmpeg`, and then if I was able to add go2rtc back to Windows as a network camera, I would be able to use it in Teams.

**Comment by felipecrs** [comment]
Another nice use case would be to allow using any arbitrary camera from go2rtc, like cameras plugged in a different computer, in any video conferencing application like Teams or Zoom.

**Comment by memphisdj** [comment]
I too am interested in having go2rtc act as ONVIF compliant source and to expose the streams in it over ONVIF. My use case is to add a stream from go2rtc into Hikvision DVR, which allows only ONVIF IP Cameras to be added.

**Comment by AlexxIT** [comment]
ONVIF server supported long time ago

**Comment by felipecrs** [comment]
https://github.com/AlexxIT/go2rtc?tab=readme-ov-file#source-onvif:~:text=to%20go2rtc%20config-,Hass%20%3E%20Settings,-%3E%20Integrations%20%3E%20Add%20Integration

**Comment by memphisdj** [comment]
> https://github.com/AlexxIT/go2rtc?tab=readme-ov-file#source-onvif:~:text=to%20go2rtc%20config-,Hass%20%3E%20Settings,-%3E%20Integrations%20%3E%20Add%20Integration

In first place sorry for hijacking your issue, but the doc you linked is if i would like to interogate my camera from go2rtc through onvif. That is not what i want. I mean each camera in go2rtc to be able to be added to another device as onvif source for it. As pictured above: IP Cam already in go2rtc -> go2rtc exposing its stream info as onvif source -> Hikvision DVR (or any device for that matter).

**Comment by felipecrs** [comment]
Ops, sorry, wrong link. I meant this:

https://github.com/AlexxIT/go2rtc?tab=readme-ov-file#module-hass

![image](https://github.com/user-attachments/assets/bb3e659c-2814-4740-a0ce-9eef5cec2565)

**Comment by AlexxIT** [comment]
Yes. This is not documented feature. It should be in the `Module: ONVIF` section. This feature works out of box and doesn't have additional settings.

**Comment by memphisdj** [comment]
Hi, thanks for pointing to the doc.

I played around a bit, had to setup rtsp auth in go2rtc, because the dvr requires it.
I changed the log level of go2rtc to debug and this is what i get: 

`<html>
<body>

5:20:22.026 PM | debug | [onvif] unsupported request:<?xml version="1.0" encoding="UTF-8" ?><soap:Envelope xmlns:soap="http://www.w3.org/2003/05/soap-envelope" xmlns:tds="http://www.onvif.org/ver10/device/wsdl" xmlns:tt="http://www.onvif.org/ver10/schema"><soap:Header><wsse:Security soap:mustUnderstand="true" xmlns:wsse="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-secext-1.0.xsd"><wsse:UsernameToken wsu:Id="UsernameToken-35" xmlns:wsu="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-utility-1.0.xsd"><wsse:Username>admin</wsse:Username><wsse:Password Type="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-username-token-profile-1.0#PasswordDigest">Um609QzCBAdQrPJW7tpEUSLWThk=</wsse:Password><wsse:Nonce EncodingType="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-soap-message-security-1.0#Base64Binary">UvUznxjrT3Gzx/dn6i5jwA==</wsse:Nonce><wsu:Created>2024-08-07T14:20:22Z</wsu:Created></wsse:UsernameToken></wsse:Security></soap:Header><soap:Body><tds:SetSystemDateAndTime><tds:DateTimeType>Manual</tds:DateTimeType><tds:DaylightSavings>true</tds:DaylightSavings><tds:TimeZone><tt:TZ>EET-2</tt:TZ></tds:TimeZone><tds:UTCDateTime><tt:Time><tt:Hour>15</tt:Hour><tt:Minute>20</tt:Minute><tt:Second>21</tt:Second></tt:Time><tt:Date><tt:Year>2024</tt:Year><tt:Month>8</tt:Month><tt:Day>7</tt:Day></tt:Date></tds:UTCDateTime></tds:SetSystemDateAndTime></soap:Body></soap:Envelope>
-- | -- | --
5:20:22.029 PM | debug | [onvif] unsupported request:<?xml version="1.0" encoding="UTF-8" ?><soap:Envelope xmlns:soap="http://www.w3.org/2003/05/soap-envelope" xmlns:tds="http://www.onvif.org/ver10/device/wsdl" xmlns:tt="http://www.onvif.org/ver10/schema"><soap:Header><wsse:Security soap:mustUnderstand="true" xmlns:wsse="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-secext-1.0.xsd"><wsse:UsernameToken wsu:Id="UsernameToken-35" xmlns:wsu="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-wssecurity-utility-1.0.xsd"><wsse:Username>admin</wsse:Username><wsse:Password Type="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-username-token-profile-1.0#PasswordDigest">kl8mDmQUrD7NX4Rk3+zG5WC2klI=</wsse:Password><wsse:Nonce EncodingType="http://docs.oasis-open.org/wss/2004/01/oasis-200401-wss-soap-message-security-1.0#Base64Binary">JH1pGhe0QPe3Gc+NteqOOg==</wsse:Nonce><wsu:Created>2024-08-07T15:20:20Z</wsu:Created></wsse:UsernameToken></wsse:Security></soap:Header><soap:Body><tds:GetServices><tds:IncludeCapability>false</tds:IncludeCapability></tds:GetServices></soap:Body></soap:Envelope>


</body>
</html>`

**Comment by felipecrs** [comment]
@memphisdj please open a separate issue.

**Comment by memphisdj** [comment]
Done, totally fair, sorry !

**Comment by bleedinge** [comment]
Also looking for a way to use a go2rtc camera in MS Teams and other camera solutions. Any plans in the near future to get this feature in?

**Comment by AlexxIT** [comment]
There are plans. But I wouldn't describe them as a near future.

**Comment by bleedinge** [comment]
Thanks for answering. I'm ok if "near future" happens in the coming months.
